polyval/src/field/backend.rs

```diff
@@ -21,12 +21,12 @@ use core::ops::Add;
         target_feature = "sse4.1",
         any(target_arch = "x86", target_arch = "x86_64")
     ),
-    feature = "insecure-soft"
+    feature = "soft"
 )))]
 compile_error!(
     "no backends available! On x86/x86-64 platforms, enable intrinsics with \
      RUSTFLAGS=\"-Ctarget-cpu=sandybridge -Ctarget-feature=+sse2,+sse4.1\" or \
-     enable **INSECURE** portable emulation with the `insecure-soft` feature"
+     enable portable emulation with the `soft` Cargo feature"
 );
```

Comment:
I wonder if we can remove this compilation error and together with the `soft` feature. Yes, there is a risk that users will not enable the necessary target features, but since you enable the `soft` feature by default I don't think this compilation error will be seen in practice.

Reply:
I think I'd rather keep the warning, or make the `soft` backend mandatory (i.e. not a cargo feature), which I'd be ok with now.

I originally added a similar warning to curve25519-dalek after getting very confused what was happening only to discover I didn't enable any of the backends.

Reply:
>make the soft backend mandatory

Yes, I think it will be the simplest option right now, i.e. use software backend by default and switch to a more efficient only if necessary target features are enabled. It will be also consistent with what we do in the `aes` crate.

Reply:
Made the `soft` feature, which should certainly help simplify adding runtime feature detection if nothing else.